checkov/terraform/checks/resource/aws/DocDBTLS.py

```diff
@@ -0,0 +1,23 @@
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceCheck
+
+
+class DocDBTLS(BaseResourceCheck):
```

Comment:
i think inheriting from `BaseResourceValueCheck`
like:
https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/aws/SQSQueueEncryption.py
might be a better fit

Reply:
Hello,

I did want to use `BaseResourceValueCheck` and through this I found it may be best to use `BaseResourceNegativeValueCheck`. Because we want to fail this check if and only if the TLS config exists and is set to 'disabled'.

However, when you look at the specific data that gets used to implement the configuration it doesn't appear to match up.

In this case, the configuration is exposed through a list of KVPs which all look like this:
```
{
    name: "name-value",
    value: "value-value"
}
```
And so you can never find the tls configuration by name as it isn't exposed as a key anywhere. So the way to find if it exists is to iterate over all objects and check: 

```
if and only if:
    conf.parameter[X].name[0] is "tls" 
then check if:
    conf.parameter[X].value[0] is "disabled"
then:
    fail
# hcl2 input is parsing name to have single entry list though it is just a string value (TypeString in tf) though this doesn't effect the check
```

I don't think the Negative Check class provides the structure to check for this type of thing, and since this is a fairly uncommon way for terraform to expose configuration so I don't know if you would want to make an abstraction to handle in a slicker manner.
For example - DyanmoDB attributes are exposed in the same way (with slightly different key names): https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/data_source_aws_dynamodb_table.go - but it is still mostly uncommon I think.

So, I think my implementation (or something pretty close) is actually the best for this check at this time given all the above.

Happy to try to refactor if i've missed something obvious!

Reply:
I agree, this check requires 2 attributes:
`if elem["name"][0] == "tls" and elem["value"][0] == "disabled":`

Reply:
thanks @nimrodkor for the help!